package.json

```diff
@@ -3,19 +3,25 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@tanstack/react-query": "4.32.6",
+    "camelize-ts": "3.0.0",
     "react": "17.0.2",
     "react-dom": "17.0.2",
     "react-router-dom": "5.2.0",
     "styled-components": "5.3.6"
   },
   "devDependencies": {
+    "@babel/plugin-proposal-private-property-in-object": "7.21.11",
     "@testing-library/jest-dom": "5.16.5",
     "@testing-library/react": "12.1.5",
+    "@testing-library/react-hooks": "8.0.1",
     "@testing-library/user-event": "14.4.3",
     "@types/jest": "29.5.3",
     "@types/node": "16.18.37",
     "@types/react": "17.0.62",
     "@types/react-dom": "17.0.17",
+    "@types/react-router-dom": "5.3.3",
+    "msw": "1.3.3",
```

Comment:
I don't think we currently use `msw` in the frontend repo, but since it's using a centralised method for getting requests which this repo doesn't have, I've used it here just for easier API call testing